modules/arrow/src/geoarrow/convert-geojson-feature-to-arrow.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  Float64 as ArrowFloat,
+  FixedSizeList,
+  Field as ArrowField,
+  List as ArrowList,
+  Vector as ArrowVector,
+  makeBuilder
+} from 'apache-arrow';
+import {
+  Point,
+  MultiPoint,
+  LineString,
+  MultiLineString,
+  Polygon,
+  MultiPolygon,
+  Feature,
+  GeoJsonObject
+} from 'geojson';
+
+/**
+ * Arrow Field and geometry vector
+ */
+export type GeoArrowReturnType = {
+  field: ArrowField;
+  geometry: ArrowVector;
+};
+
+// type guard to check if feature is a GeoJSON Feature or Geometry
+export function isFeature(json: unknown): json is Feature {
+  return Boolean((json as Feature).type === 'Feature');
+}
+
+/**
+ * Convert GeoJson Features to GeoArrow
+ * @param name
+ * @param features
+ * @returns GeoArrowReturnType
+ */
+export function geojsonFeaturesToArrow(
+  name: string,
+  features: Array<Feature | GeoJsonObject>
+): GeoArrowReturnType {
+  // get the geometry type by iterating all features
+  // if one geometry is multi-, then the whole geometry is multi-
+  let geometryType = '';
+  for (let i = 0; i < features.length; i++) {
+    const feature = features[i];
+    const geometry = isFeature(feature) ? feature.geometry : feature;
+    if (i === 0) {
+      geometryType = geometry.type;
+    }
+    if (geometry.type.includes('Multi')) {
```

Comment:
Maybe more readable?
```suggestion
    if (geometry.type.startsWith('Multi')) {
```